src/QsCompiler/Core/DeclarationHeaders.fs

```diff
@@ -108,24 +108,55 @@ module DeclarationHeader =
         Serializer.Serialize(new StringWriter(builder), obj)
         builder.ToString()
 
+/// <summary>
+/// The schema for <see cref="TypeDeclarationHeader"/> that is used with JSON serialization.
+/// </summary>
+[<CLIMutable>]
+[<DataContract>]
+type private TypeDeclarationHeaderSchema =
+    {
+        [<DataMember>]
+        QualifiedName: QsQualifiedName
+        [<DataMember>]
+        Attributes: ImmutableArray<QsDeclarationAttribute>
+        [<DataMember>]
+        Modifiers: Modifiers
+        [<DataMember>]
+        SourceFile: string
+        [<DataMember>]
+        Position: DeclarationHeader.Offset
+        [<DataMember>]
+        SymbolRange: DeclarationHeader.Range
+        [<DataMember>]
+        Type: ResolvedType
+        [<DataMember>]
+        TypeItems: QsTuple<QsTypeItem>
+        [<DataMember>]
+        Documentation: ImmutableArray<string>
+    }
 
 /// to be removed in future releases
 type TypeDeclarationHeader =
     {
         QualifiedName: QsQualifiedName
         Attributes: ImmutableArray<QsDeclarationAttribute>
         Modifiers: Modifiers
-        SourceFile: string
+        Source: Source
         Position: DeclarationHeader.Offset
         SymbolRange: DeclarationHeader.Range
         Type: ResolvedType
         TypeItems: QsTuple<QsTypeItem>
         Documentation: ImmutableArray<string>
     }
+
     [<JsonIgnore>]
     member this.Location = DeclarationHeader.CreateLocation(this.Position, this.SymbolRange)
 
-    member this.FromSource source = { this with SourceFile = source }
+    // TODO: RELEASE 2021-07: Remove TypeDeclarationHeader.SourceFile.
```

Comment:
Just curious about why is this being removed in 2021-07 release. Is this because it is six months after the release of this change?

Reply:
Yes, I believe the compiler API tries to follow a 6-month deprecation period when possible.